types/validator_set.go

```diff
@@ -632,8 +632,14 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID,
 	if vals.Size() != len(commit.Signatures) {
 		return NewErrInvalidCommitSignatures(vals.Size(), len(commit.Signatures))
 	}
-	if err := verifyCommitBasic(commit, height, blockID); err != nil {
-		return err
+
+	// Validate Height and BlockID.
+	if height != commit.Height {
+		return NewErrInvalidCommitHeight(height, commit.Height)
```

Comment:
Why `New` here?

Reply:
you mean why create a function instead of constructing a struct here?

Reply:
No, why is the error prefixed with `New`?

Reply:
it's a function, not an error itself

Reply:
Oh, of course 🤦‍♂️